src/js/transfers/bridged-nep21-to-erc20/index.js

```diff
@@ -6,26 +6,44 @@ import { borshifyOutcomeProof } from './borshify-proof'
 import { COMPLETE, FAILED, SUCCESS } from '../statuses'
 import * as storage from '../storage'
 
-// Call contract given by `nep21Address` to withdraw (burn) the given `amount`
-// of NEP21 tokens for window.nearUserAddress. The raw transaction info is
-// returned. This info is then used in `checkStatus` to derive the proof needed
-// for releasing `amount` tokens to window.ethUserAddress on Ethereum.
+// Immediately return to caller, which adds an ID to the provided data, stores
+// it in localStorage, and then calls `checkStatus` (below), which will then
+// call actual `withdraw` function.
 export async function initiate (nep21Address, amount) {
+  return {
+    amount,
+    nep21Address,
+    status: INITIALIZED
+  }
+}
+
+function withdraw ({ id, nep21Address, amount }) {
   const bridgeToken = new NearContract(
     window.nearConnection.account(),
     nep21Address,
     { changeMethods: ['withdraw'] }
   )
 
-  // TODO: can cause redirect to NEAR Wallet, rendering `tx` inaccessible!
-  const tx = await bridgeToken.withdraw(
-    {
-      amount: String(amount),
-      recipient: window.ethUserAddress.replace('0x', '')
-    },
-    new BN('3' + '0'.repeat(14)) // 10x current default from near-api-js
-  )
+  // withdraw transaction causes a redirect to NEAR Wallet, so we use
+  // 'sendTransaction' to add it to a queue, along with some metadata to
+  // identify the transfer when we return to this app. This is processed via
+  // the `inFlightTransactions` logic in authNear, which will call
+  // `processWithdrawTx`
```

Comment:
so my vision for this was that you just do:

```
bridgetoken.withdraw({
           amount: String(amount),
           recipient: window.ethUserAddress.replace('0x', '')
         },
         {
           gas: new BN('3' + '0'.repeat(14)) // 10x current default from near-api-js,
           meta: { id }
         }
);
```

(basically the queue logic should be setup by default when you use WalletConnection)